RedHat-UX/red-hat-design-system
# feat(pagination): add open, small, and compact variants

## What I did

1. Added ~~three~~ two new attributes to pagination: `variant="open"` and `size="sm"`. Compact is now included by default in the main component between widths of 344px and 767px.

## Testing Instructions

1. Check out the deploy preview for the following variants:
    * [Open](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/open/)
    * [Open compact](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/open-compact/)
    * [Open compact small](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/open-compact-size/)
    * [Small](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/size/)
    * [Small compact](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/size-compact/)
    * [Compact](https://deploy-preview-1587--red-hat-design-system.netlify.app/elements/pagination/demo/compact/)

## To do's

- [x] Update ux.redhat.com `rh-pagination` docs with the new variants/attributes
- [x] See [more docs to do's](https://github.com/RedHat-UX/red-hat-design-system/pull/1587#issuecomment-2197263029)
- [x] Change button dimensions for default size to 48px in Figma library to stick with token values.

## Notes to Reviewers

Relies on style changes made to pagination in #1552. Solves most of the to do's in #1457. 

### Pagination + items per page selector (blocked)

The [Figma spec](https://www.figma.com/file/BVTQ5Fim2mLfstNGeZN1Qc/Pagination?type=design&node-id=0%3A1&mode=design&t=mTimrsdl2tizInMQ-1) denotes another variant called "Pagination + items per page selector". This items per page selector relies on a dropdown menu. Since [rh-dropdown doesn't exist yet](https://github.com/RedHat-UX/red-hat-design-system/issues/425), that variant is blocked.
